// react-context-devtool/dist/module/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.debugContextDevtool = exports.ContextDevTool = void 0;

var _react = _interopRequireDefault(require("react"));

let debugOptions = {
  debugReducer: true,
  debugContext: true,
  disable: false,
  disableAutoMode: false
};

const ContextDevTool = ({
  id,
  context: Context,
  displayName
}) => {
  if (debugOptions.disable) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement(Context.Consumer, null, values => {
    if (typeof window !== "undefined" && window._REACT_CONTEXT_DEVTOOL) {
      window._REACT_CONTEXT_DEVTOOL({
        id,
        displayName,
        values
      });
    }

    return null;
  });
};

exports.ContextDevTool = ContextDevTool;

const debugContextDevtool = (container, options) => {
  debugOptions = { ...debugOptions,
    ...options
  };

  if (debugOptions.disable || debugOptions.disableAutoMode || typeof window === "undefined" || !debugOptions.debugReducer && !debugOptions.debugContext) {
    return false;
  }

  const hook = window.__REACT_CONTEXT_DEVTOOL_GLOBAL_HOOK;

  if (hook && hook.debugFiber) {
    if (hook.helpers && hook.helpers.loadHookHelper && debugOptions.debugReducer) {
      hook.helpers.loadHookHelper().then(() => {
        hook.debugFiber(container, options);
      });
    } else {
      hook.debugFiber(container, options);
    }
  }
};

exports.debugContextDevtool = debugContextDevtool;
